scheduler/scheduler_system.go

```diff
@@ -279,15 +279,38 @@ func (s *SystemScheduler) computeJobAllocs() error {
 	return s.computePlacements(diff.place)
 }
 
+func mergeNodeFiltered(acc, curr *structs.AllocMetric) *structs.AllocMetric {
+	if acc == nil {
+		v := *curr
+		return &v
```

Comment:
Shallow copying means referenced values in `curr` on the first call will be mutated on successive calls.